src/Enterspeed.Source.UmbracoCms/Services/DataProperties/DefaultConverters/DefaultBlockGridPropertyValueConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿#if NET7_0_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Enterspeed.Source.Sdk.Api.Models.Properties;
+using Enterspeed.Source.UmbracoCms.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Cms.Core.Models.Blocks;
+using Umbraco.Cms.Core.Models.PublishedContent;
+
+namespace Enterspeed.Source.UmbracoCms.Services.DataProperties.DefaultConverters
+{
+    public class DefaultBlockGridPropertyValueConverter : IEnterspeedPropertyValueConverter
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public DefaultBlockGridPropertyValueConverter(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public bool IsConverter(IPublishedPropertyType propertyType)
+        {
+            return propertyType.EditorAlias.Equals("Umbraco.BlockGrid");
+        }
+
+        public virtual IEnterspeedProperty Convert(IPublishedProperty property, string culture)
+        {
+            var value = property.GetValue<BlockGridModel>(culture);
+            var arrayItems = new List<IEnterspeedProperty>();
+
+            if (value != null)
+            {
+                // NOTE: This needs to be resolved manually, since it would cause a circular dependency if injected through constructor
+                var dataPropertyService = _serviceProvider.GetRequiredService<IEnterspeedPropertyService>();
+
+                foreach (var item in value)
+                {
+                    var properties = new Dictionary<string, IEnterspeedProperty>();
+
+                    if (item.Areas.Any())
+                    {
+                        var areas = AddAreas(item, dataPropertyService);
+                        properties.Add("areas", areas);
+                    }
+                    else
+                    {
+                        if (item.Content?.Properties != null)
+                        {
+                            var contentProperties = dataPropertyService.ConvertProperties(item.Content.Properties, culture);
+                            properties.Add("content", new ObjectEnterspeedProperty(contentProperties));
+                        }
+
+                        if (item.Settings?.Properties != null)
+                        {
+                            var settingsProperties = dataPropertyService.ConvertProperties(item.Settings.Properties, culture);
+                            properties.Add("settings", new ObjectEnterspeedProperty(settingsProperties));
+                        }
+
```

Comment:
I also think we need to map some of the additional properties about the columns and rows in order for the frontend to have the needed information to render the grid. I have marked some, but there might be others, e.g. the number of columns for the grid it self
![image](https://user-images.githubusercontent.com/2575817/210247790-0fbd0a24-6167-4321-acfe-670041a359ab.png)

Reply:
These has now been added. :)